src/entities/Snapshot/utils.ts

```diff
@@ -1,9 +1,39 @@
 import { Wallet } from '@ethersproject/wallet';
-import { requiredEnv } from 'decentraland-gatsby/dist/utils/env';
+import { intersection } from 'lodash';
+import { SnapshotVote } from '../../api/Snapshot';
 
-const SNAPSHOT_PRIVATE_KEY = requiredEnv('SNAPSHOT_PRIVATE_KEY')
-export const SNAPSHOT_ACCOUNT = new Wallet(SNAPSHOT_PRIVATE_KEY)
+export interface MatchResult {
+  percentage: number,
+  matches: { proposal_id: string, sameVote: boolean }[]
+}
 
 export async function signMessage(wallet: Wallet, msg: string) {
   return wallet.signMessage(Buffer.from(msg, 'utf8'));
 }
+
+export function calculateMatch(votes1: SnapshotVote[], votes2: SnapshotVote[]): MatchResult {
+
+  const match: MatchResult = { percentage: 0, matches: [] }
+  const commonProposalIds = intersection(votes1.map(vote => vote.proposal.id), votes2.map(vote => vote.proposal.id))
+
+  if (commonProposalIds.length !== 0) {
+
+    let matchCounter = 0
+    for (const proposalId of commonProposalIds) {
+      const vote1 = votes1.find(v => v.proposal.id === proposalId)!
+      const vote2 = votes2.find(v => v.proposal.id === proposalId)!
```

Comment:
I think maybe we can avoid calculating the intersection and the two finds if we just make one pass on one of the vote arrays, and look for matches in the other:

```typescript
export function calculateMatch(votes1: SnapshotVote[], votes2: SnapshotVote[]): MatchResult {
  const match: MatchResult = { percentage: 0, matches: [] }

  let matchCounter = 0
  let proposalsInCommon = 0
  for (const vote1 of votes1) {
    const proposalId = vote1.proposal.id
    const vote2 = votes2.find((v) => v.proposal.id === proposalId)
    if (vote2) {
      proposalsInCommon++
      if (vote1.choice === vote2.choice) {
        matchCounter++
        match.matches.push({ proposal_id: proposalId, sameVote: true })
      } else {
        match.matches.push({ proposal_id: proposalId, sameVote: false })
      }
    }
  }
  if (proposalsInCommon > 0) {
    match.percentage = Math.round((matchCounter / proposalsInCommon) * 100)
  }
  return match
}
```